src/topology/algebra/ring.lean

```diff
@@ -150,51 +154,50 @@ open ideal.quotient
 instance topological_ring_quotient_topology : topological_space N.quotient :=
 by dunfold ideal.quotient submodule.quotient; apply_instance
 
+-- note for the reader: in the following, `mk` is `ideal.quotient.mk`, the canonical map `R → R/I`.
+
+/--
+The preimage of the image of a set under the quotient map is the union of the cosets covered by `s`.
+-/
 lemma quotient_ring_saturate {α : Type*} [comm_ring α] (N : ideal α) (s : set α) :
   mk N ⁻¹' (mk N '' s) = (⋃ x : N, (λ y, x.1 + y) '' s) :=
 begin
   ext x,
   simp only [mem_preimage, mem_image, mem_Union, ideal.quotient.eq],
-  split,
-  { exact assume ⟨a, a_in, h⟩, ⟨⟨_, N.neg_mem h⟩, a, a_in, by simp⟩ },
-  { exact assume ⟨⟨i, hi⟩, a, ha, eq⟩, ⟨a, ha,
-      by rw [← eq, sub_add_eq_sub_sub_swap, sub_self, zero_sub];
-      exact N.neg_mem hi⟩ }
+  exact ⟨λ ⟨a, a_in, h⟩, ⟨⟨_, N.neg_mem h⟩, a, a_in, by simp⟩,
+         λ ⟨⟨i, hi⟩, a, ha, eq⟩,
+           ⟨a, ha, by rw [← eq, sub_add_eq_sub_sub_swap, sub_self, zero_sub]; exact N.neg_mem hi⟩⟩
 end
+-- TODO: Move this result to a more appropriate place; this doesn't require `α` to be topological.
+-- Also, any ideas for a name?
 
 variable [topological_ring α]
 
 lemma quotient_ring.is_open_map_coe : is_open_map (mk N) :=
 begin
-  assume s s_op,
-  show is_open (mk N ⁻¹' (mk N '' s)),
-  rw quotient_ring_saturate N s,
-  exact is_open_Union (assume ⟨n, _⟩, is_open_map_add_left n s s_op)
+  intros s s_op,
+  change is_open (mk N ⁻¹' (mk N '' s)),
+  rw quotient_ring_saturate,
+  exact is_open_Union (λ ⟨n, _⟩, is_open_map_add_left n s s_op)
 end
 
 lemma quotient_ring.quotient_map_coe_coe : quotient_map (λ p : α × α, (mk N p.1, mk N p.2)) :=
-begin
-  apply is_open_map.to_quotient_map,
-  { exact (quotient_ring.is_open_map_coe N).prod (quotient_ring.is_open_map_coe N) },
-  { exact (continuous_quot_mk.comp continuous_fst).prod_mk
-          (continuous_quot_mk.comp continuous_snd) },
-  { rintro ⟨⟨x⟩, ⟨y⟩⟩,
-    exact ⟨(x, y), rfl⟩ }
-end
+is_open_map.to_quotient_map
+((quotient_ring.is_open_map_coe N).prod (quotient_ring.is_open_map_coe N))
+((continuous_quot_mk.comp continuous_fst).prod_mk (continuous_quot_mk.comp continuous_snd))
+(by rintro ⟨⟨x⟩, ⟨y⟩⟩; exact ⟨(x, y), rfl⟩)
+-- `rintro` seems to be able to deal with `quotient` whilst `λ` can't :(
```

Comment:
Why leaving such a comment in the file?
```suggestion
```

Reply:
was basically a "note to reviewers" in case I was missing something obvious